src/libraries/Microsoft.PowerFx.Core/Public/Values/RecordValue.cs

```diff
@@ -26,7 +26,7 @@ public abstract class RecordValue : ValidFormulaValue, IMutationCopy
         /// Fields and their values directly available on this record. 
         /// The field names should match the names on <see cref="Type"/>. 
         /// </summary>
-        public IEnumerable<NamedValue> Fields => GetFields();
+        public virtual IEnumerable<NamedValue> Fields => GetFields();
```

Comment:
>virtual [](http://example.com/codeflow?start=15&length=7)

This is a concern.

We get these from the type, which has the potential virtual. 
We shouldn't have virtual fields per-value.

Reply:
Removed, was for initial testingn only